compiler/rustc_span/src/span_encoding.rs

```diff
@@ -71,7 +71,7 @@ use rustc_data_structures::fx::FxIndexSet;
 /// the dependency to the parent definition's span. This is performed
 /// using the callback `SPAN_TRACK` to access the query engine.
 ///
-#[derive(Clone, Copy, Eq, PartialEq, Hash)]
+#[derive(Clone, Copy, Eq, PartialEq, Hash, TypeFoldable, TypeVisitable)]
```

Comment:
why is that needed?

Reply:
Appears in various tuple fields of different types and some tuple arguments, where tuple is not noop traversable but field/argument is required to be traversable.